components/shared/profile/edit/EditTech.tsx

```diff
@@ -0,0 +1,113 @@
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import React from "react";
+
+interface EditTechProps {
+  technologies: any;
+  techStackBadges: any;
+  handleRemoveTech: any;
+  search: any;
+  results: any;
+  setValue: any;
+  setSearch: any;
+}
+const EditTech = ({
+  techStackBadges,
+  technologies,
+  handleRemoveTech,
+  search,
+  setValue,
+  results,
+  setSearch,
+}: EditTechProps) => {
```

Comment:
```suggestion
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import React, { useCallback, useEffect, useState } from "react";
import { techStackBadges } from "@/lib/constants";

interface EditTechProps {
  technologies: string[] | undefined;
  setValue: (name: string, value: any) => void;
}
const EditTech = ({ technologies, setValue }: EditTechProps) => {
```

Our component can just look like this. None of anything else is really getting used elsewhere, so there's no reason to define it in 'EditProfile' instead of in this component.

For instance 'handleRemoveTech'. The only time we're calling that is here in this component, so we shouldn't be prop drilling it down into this component, it should just be defined here.

Same for the useEffect that controls the search results. This component is the only one that cares about the search results- so why handle that logic anywhere but right inside this component?